### PR TITLE
Update schema

### PR DESCRIPTION
**Pull request by AlmightyYakob**
In response to #103.

Hopefully this covers most of it, but there's sometimes some ambiguity in the [definition](https://github.com/BICCN/BICCN-Infrastructure-Draft/blob/ce0df09b310bbcdd0a7a63cc51a0955e3cae97b8/BICCN-policies/dataset-metadata.yaml)

Please let me know if anything is missing/incorrect

### CODE REVIEW

File: web/src/assets/schema/nwb.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "categorical": {
+          "$id": "#/properties/age/properties/categorical",
+          "type": "object",
+          "title": "Categorical"
+        }
+      }
+    },
+    "sex": {
+      "$id": "#/properties/sex",
+      "type": "string",
```
Comment (satra) [inline web/src/assets/schema/nwb.json]: this should be an array of strings

File: web/src/assets/schema/base.json [hunk 2]
```diff
@@ -284,7 +258,7 @@
         }
       }
     },
-    "ethics_approvals": {
+    "ethicsApprovals": {
```
Comment (satra) [inline web/src/assets/schema/base.json]: should be an array of these objects.

File: web/src/assets/schema/base.json [hunk 3]
```diff
@@ -376,6 +350,69 @@
       "$id": "#/properties/language",
       "type": "string",
       "title": "Language"
+    },
+    "associatedData": {
+      "$id": "#/properties/associatedData",
+      "type": "object",
```
Comment (satra) [inline web/src/assets/schema/base.json]: also an array of objects.

File: web/src/assets/schema/base.json [hunk 4]
```diff
@@ -108,33 +127,6 @@
...
-        "affiliations": {
-          "$id": "#/properties/contact/properties/affiliations",
-          "type": "string",
-          "title": "Affiliations"
-        }
-      }
-    },
     "sponsors": {
       "$id": "#/properties/sponsors",
       "type": "object",
```
Comment (satra) [inline web/src/assets/schema/base.json]: sponsors should also be an array

**Comment by satra** [comment]
@AlmightyYakob - this looks good. i left some comments on things that should be arrays of objects. if encoded a specific object versus an array in the yaml using the list syntax. so wherever there is a `-` it means that key should be in an array.

for example:
```
- key1:
  - key2:
     key3:
```
so `key1` is an array of objects with keys `key2` and `key3` 

the one other thing is to add to `sponsors` section an optional field called `Award Number` this would be a string.

**Comment by AlmightyYakob** [comment]
@satra thanks for the feedback, I'm fixing that now. It also seems like these properties should be arrays of objects as well: 

https://github.com/BICCN/BICCN-Infrastructure-Draft/blob/ce0df09b310bbcdd0a7a63cc51a0955e3cae97b8/BICCN-policies/dataset-metadata.yaml#L16-L24

Since they're formatted the same way as the ones you just mentioned. Is that correct?

**Comment by satra** [comment]
> Since they're formatted the same way as the ones you just mentioned. Is that correct?

yes

i just noticed i made a mistake with respect to that syntax for age in the nwb section. that should not be a list but either an object. fixed  here: https://github.com/BICCN/BICCN-Infrastructure-Draft/blob/3f07606f7a07584d35e6ff097ff9ecda17f5b7fc/BICCN-policies/dataset-metadata.yaml

**Comment by AlmightyYakob** [comment]
@satra I made changes, hopefully now it's accurate

**Comment by satra** [comment]
@AlmightyYakob - looks good now.

File: web/src/assets/schema/base.json [hunk 5]
```diff
@@ -108,92 +127,52 @@
...
```
Comment (satra) [inline web/src/assets/schema/base.json]: sponsors should be an array.

also we should a string field to the object called `Award number`

Comment (AlmightyYakob) [reply web/src/assets/schema/base.json]: > sponsors should be an array.

Oops, forgot to actually update the type itself. 

> also we should a string field to the object called `Award number`

To the sponsors object (the objects contained within the sponsors array)?

Comment (satra) [reply web/src/assets/schema/base.json]: yes, another field to the sponsors object:

```
- sponsors:
   - name:
     ...
     award_number:
```

**Comment by AlmightyYakob** [comment]
Should be good to go

**Comment by satra** [comment]
This came up in the meeting today and had skipped my mind. Can we change "Associated Disease" to "Conditions studied"?

**Pull request merged**